src/source/Ice/IceAgent.c

```diff
@@ -695,6 +715,145 @@ STATUS iceAgentShutdown(PIceAgent pIceAgent)
     return retStatus;
 }
 
+STATUS iceAgentRestart(PIceAgent pIceAgent, PCHAR localIceUfrag, PCHAR localIcePwd)
+{
+    ENTERS();
+
+    STATUS retStatus = STATUS_SUCCESS;
+    BOOL locked = FALSE;
+    PDoubleListNode pCurNode = NULL, pNextNode = NULL;
+    PIceCandidate pLocalCandidate = NULL;
+    PIceCandidatePair pIceCandidatePair = NULL;
+    UINT32 i;
+    ATOMIC_BOOL alreadyRestarting;
+    PIceCandidate localCandidates[100];
```

Comment:
100 can be replaced with `KVS_ICE_MAX_LOCAL_CANDIDATE_COUNT`

Reply:
fixed